### PR TITLE
Single file share

### PR DESCRIPTION
**Pull request by L-Henke**
Added externalFile table and owner to FileMetadata

Changes implemented in https://github.com/Qabel/qabel-android/pull/230

### CODE REVIEW

File: _docs/Qabel-Protocol-Box.md [hunk 1]
```diff
@@ -281,6 +285,19 @@ url: STR // URL to the metadata file that contains information about the folder
...
 ```
 
+ExternalFile:
+
+```
+{
+owner: STR, // public key of the owner of that File
+name: STR, // object name,
+size: LONG, // uncompressed file size
+mtime: LONG, // modification time as seconds since epoch
```
Comment (julianseeger) [inline _docs/Qabel-Protocol-Box.md]: How can you track mtime and size changes? They would only reach the FM wouldn't they?

Comment (L-Henke) [inline _docs/Qabel-Protocol-Box.md]: I guess one would have to refresh the external files regularly and update these information. Additionally, this information would only reach the FM if the creator of the share would update them regularly.

Comment (roeslpa) [inline _docs/Qabel-Protocol-Box.md]: The FM in the creators prefix is automatically updated during "Updating a file". Isn't it sufficient to only store owner, meta, block and maybe name and request current size and modification time from the FM?

File: _docs/Qabel-Protocol-Box.md [hunk 2]
```diff
@@ -274,6 +275,7 @@ External:
 
 ```
 {
+is_folder: BOOL, // indicates if external is a folder or a file
```
Comment (roeslpa) [inline _docs/Qabel-Protocol-Box.md]: Could you add the type definition of `BOOL` in [this file](https://github.com/Qabel/qabel.github.io/blob/master/_docs/Qabel-Client-Local-Data.md), please?

**Comment by roeslpa** [comment]
Except from the comment above I ack this :+1:

File: _docs/Qabel-Protocol-Box.md (hunk 2 above)
Comment (L-Henke) [inline _docs/Qabel-Protocol-Box.md]: Done

**Pull request merged**